src/traces/scatter/arrays_to_calcdata.js

```diff
@@ -15,6 +15,9 @@ var Lib = require('../../lib');
 // arrayOk attributes, merge them into calcdata array
 module.exports = function arraysToCalcdata(cd, trace) {
 
+    // so each point knows which index it originally came from
+    for(var i = 0; i < cd.length; i++) cd[i].i = i;
```

Comment:
Good :eye: here and nice fix. But, I think the _solution_ would be to make `Lib.mergeArray` sanatize those arrays instead of :hocho: it. We can debate this in a future issue / PR :smile: